packages/compass-indexes/src/components/create-index-fields.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useCallback, useMemo } from 'react';
+import { hasColumnstoreIndexesSupport } from '../utils/has-columnstore-indexes-support';
+
+import {
+  Combobox,
+  ComboboxOption,
+  Select,
+  Option,
+  spacing,
+  css,
+  ListEditor,
+} from '@mongodb-js/compass-components';
+
+import type { IndexField } from '../modules/create-index/fields';
+
+/**
+ * Current allowed types for indexes.
+ */
+const INDEX_TYPES = ['1 (asc)', '-1 (desc)', '2dsphere', 'text', 'columnstore'];
+
+/**
+ * Default values for field name and type as presented in the UI.
+ */
+const DEFAULT_FIELD = {
+  name: 'Select or type a field name',
+  type: 'Select a type',
+};
+
+const createIndexFieldsStyles = css({
+  display: 'flex',
+  gap: spacing[2],
+});
+
+const createIndexFieldsNameStyles = css({
+  flexGrow: 1,
+  textTransform: 'none',
+  whiteSpace: 'nowrap',
+});
+
+const createIndexFieldsTypeStyles = css({
+  flexGrow: 1,
```

Comment:
Probably better to set a fixed `width` for this one. Flex causes some weird layout as the type field is not always the same size:

<img width="550" alt="Screenshot 2022-10-18 at 09 17 14" src="https://user-images.githubusercontent.com/334881/196362948-f46f05f9-ddf3-40d8-9fbd-4e42eaa2dafa.png">

Reply:
Ah nice catch. Updated.